### ISSUE TITLE
"unknown flag: --s2i" while running odo test suite 'test-generic' on ppc64le

### ISSUE THREAD

**Issue by sarveshtamba** [openshift/odo#3934]
I pulled the latest odo master branch and ran the odo test suite 'test-generic' on ppc64le. Facing the following issue:-
```
Running odo with args [odo create --s2i nodejs nodejs --project yifjsntrvn --git https://github.com/sclorg/nodejs-ex]
[odo] Error: unknown flag: --s2i
[odo] Usage:
[odo]   odo create <component_type> [component_name] [flags]
[odo]
[odo] Examples:
[odo]   # Create a new Node.JS component with existing sourcecode as well as specifying a name
[odo]   odo create nodejs mynodejs
[odo]
[odo]   # Name is not required and will be automatically generated if not passed
[odo]   odo create nodejs
[odo]
[odo]   # List all available components before deploying
[odo]   odo catalog list components
[odo]   odo create java-quarkus
[odo]
[odo]   # Download an example devfile and application before deploying
[odo]   odo create nodejs --starter
[odo]
[odo]   # Using a specific devfile
[odo]   odo create mynodejs --devfile ./devfile.yaml
[odo]   odo create mynodejs --devfile https://raw.githubusercontent.com/odo-devfiles/registry/master/devfiles/nodejs/devfile.yaml
[odo]
[odo]   # Create new Node.js component named 'frontend' with the source in './frontend' directory
[odo]   odo create nodejs frontend --context ./frontend
[odo]
[odo]   # Create new Java component with binary named sample.jar in './target' directory
[odo]   odo create java:8  --binary target/sample.jar
[odo]
[odo]   # Create new Node.js component with source from remote git repository
[odo]   odo create nodejs --git https://github.com/openshift/nodejs-ex.git
[odo]
[odo]   # Create new Node.js component with custom ports and environment variables
[odo]   odo create nodejs --port 8080,8100/tcp,9100/udp --env key=value,key1=value1
[odo]
[odo] Flags:
[odo]       --app string       Application, defaults to active application
[odo]   -b, --binary string    Create a binary file component component using given artifact. Works only with Java components. File needs to be in the context directory.
[odo]       --context string   Use given context directory as a source for component settings
[odo]       --env strings      Environmental variables for the component. For example --env VariableName=Value
[odo]   -g, --git string       Create a git component using this repository.
[odo]   -h, --help             Help for create
[odo]       --now              Push changes to the cluster immediately
[odo]   -p, --port strings     Ports to be used when the component is created (ex. 8080,8100/tcp,9100/udp)
[odo]       --project string   Project, defaults to active project
[odo]   -r, --ref string       Use a specific ref e.g. commit, branch or tag of the git repository (only valid for --git components)
[odo]
[odo] Additional Flags:
[odo]   -o, --o string             Specify output format, supported format: json
[odo]   -v, --v Level              Number for the log level verbosity. Level varies from 0 to 9 (default 0).
[odo]       --vmodule moduleSpec   Comma-separated list of pattern=N settings for file-filtered logging
[odo]
[odo]  ✗  unknown flag: --s2i```

Attached is the detailed spec file log - 
[odo_test-generic_failures_ppc64le_2.txt](https://github.com/openshift/odo/files/5207441/odo_test-generic_failures_ppc64le_2.txt)

**Comment by sarveshtamba** [comment]
Seems to have been introduced after changes in https://github.com/openshift/odo/pull/3705/
@cdrage @johnmcollier ^^

**Comment by prietyc123** [comment]
@sarveshtamba I ran the same step on latest master and couldn't find the error.

```
$ odo version
odo v2.0.0 (3fcd3670e)

$ odo create --s2i nodejs nodejs --project yifjsntrvn --git https://github.com/sclorg/nodejs-ex --port 8080
Validation
Warning: nodejs is not fully supported by odo, and it is not guaranteed to work
 ✓  Validating component [620ms]

Please use `odo push` command to create the component with source deployed
```
Can you please check odo version once and confirm.

**Comment by prietyc123** [comment]
Did you build the master binary then ran the test suite against it?

**Comment by sarveshtamba** [comment]
@prietyc123 Yes this is the master binary. Standalone command works fine.
```
[root@shivani2-6b65-bastion odo]# ./odo version
odo v2.0.0 (3fcd3670e)

Server: https://api.shivani2-6b65.ocp-44.com:6443
Kubernetes: v1.18.3+012b3ec
[root@shivani2-6b65-bastion odo]# ./odo create --s2i nodejs nodejs --project yifjsntrvn --git https://github.com/sclorg/nodejs-ex --port 8080
Validation
 ✓  Validating component [36ms]

Please use `odo push` command to create the component with source deployed
```

**Comment by sarveshtamba** [comment]
Resolved with #3891 , closing this for now.
